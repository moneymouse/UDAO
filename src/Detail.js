import style from "./Detail.module.css"
import Gy from "./img/ninja161787932432540.jpg"
import React from "react"
import Nav from "./Nav";
import DefaultBoard from "./defaultBoard";
import {PicDetBoard, ButtonBar} from "./tempComponent";
import {BASE_URL} from "./utils/setting";
import * as echarts from "echarts";

import('./img/ninja161787932432540.jpg').then(e => console.log(e))

function combine(...a) {
	return Object.assign(...a);
}

export default class Detail extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			itemName: "",
			itemTarget: 1,
			itemNow: 0,
			itemDetail: "",
			itemInvolve: "",
			itemImg: "",
		}
	}

	componentDidMount() {
		let myCharts = echarts.init(document.getElementById('main'));
		fetch(BASE_URL + '/item/getItem?item_id=' + this.props.match.params.item_id).then(res => res.json()).then(res => {
			if (res.code == 200) {
				this.setState(res.result)
				let option = {
					tooltip: {
						trigger: 'item'
					},
					legend: {
						top: '5%',
						left: 'center'
					},
					series: [
						{
							name: '资金池',
							type: 'pie',
							radius: ['60%', '90%'],
							avoidLabelOverlap: false,
							itemStyle: {
								borderRadius: 10,
								borderColor: '#fff',
								borderWidth: 2
							},
							emphasis: {
								label: {
									show: true,
									fontSize: '40',
									fontWeight: 'bold'
								}
							},
							labelLine: {
								show: false
							},
							label: {
								show: true,
								position: 'center',
								formatter: res.result.itemNow/res.result.itemTarget+'%',
								fontSize: '12',
								fontWeight:'700'
							},
							data: [
								{value: res.result.itemNow},
								{value: res.result.itemTarget-res.result.itemNow},
							]
						}
					]
				};
				myCharts.setOption(option);
			}
			else{
				this.props.history.push('/404')
			}
		}).catch(err=>console.log(err))
	}

	render() {
		const {history} = this.props;
		return (
			<React.Fragment>
				<Nav couldback={true} histories={history}
				     onBack={() => history.push('/home')}>{this.state.itemName}</Nav>
				<img style={{
					borderRadius: "12px",
					width: "88%",
					marginLeft: "6%",
					height: "27.6vh",
					marginBottom: "2vw",
					marginTop: "4vw"
				}} src={this.state.itemImg||Gy}/>
				<PicDetBoard>
					<div style={{flex: 3,height:'10vh'}} id={'main'}></div>
					<div className={style.contributionDetail}>
						<b>现筹款数额：{this.state.itemNow}</b>
						<br/>
						<b>筹款总目标：{this.state.itemTarget}</b>
					</div>
				</PicDetBoard>
				<DefaultBoard className={style.detailBoard}>
					<div className={style.detailTitle}>项目详情</div>
					{this.state.itemDetail}
				</DefaultBoard>
				<ButtonBar onLeftClick={()=>{alert('您是内测用户，暂时无法捐赠。')}} onRightClick={() => history.push('/detail/vote/'+this.props.match.params.item_id)} leftColor={"#E56862"}
				           leftContent={"立即捐款"}
				           rightColor={"#F59B40"} rightContent={"进入投票"}/>
			</React.Fragment>
		);
	}
}


