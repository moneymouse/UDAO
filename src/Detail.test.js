import { render, screen } from '@testing-library/react';
import Detail from "./Detail";
import renderer from 'react-test-renderer';
test("detail snapshot",()=>{
	let detailPage = renderer.create(<Detail history={{goBack: () => "abc"}} />)
	let tree = detailPage.toJSON();
	expect(tree).toMatchSnapshot();
})