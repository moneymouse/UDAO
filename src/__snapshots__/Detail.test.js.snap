// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detail snapshot 1`] = `
Array [
  <div
    className="nav"
  >
    <div
      className="back"
      onClick={[Function]}
    >
      <img
        src="back.svg"
      />
    </div>
    <h2
      style={
        Object {
          "float": "left",
          "lineHeight": "10vh",
          "marginBottom": "unset",
          "marginLeft": "22.5%",
          "marginTop": 0,
          "textAlign": "center",
          "width": "55%",
        }
      }
    >
      幸存老兵助养行动
    </h2>
    <div
      className="person"
    >
      <img
        src="person.svg"
      />
    </div>
  </div>,
  <div
    style={
      Object {
        "height": "10vh",
      }
    }
  />,
  <img
    src="ninja161787932432540.jpg"
    style={
      Object {
        "borderRadius": "12px",
        "height": "27.6vh",
        "marginBottom": "2vw",
        "marginLeft": "6%",
        "marginTop": "4vw",
        "width": "88%",
      }
    }
  />,
  <div
    className="dataBoard"
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": "12px",
        "height": "unset",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 3,
        }
      }
    />
    <div
      className="contributionDetail"
    >
      <b>
        现筹款数额：
        0
      </b>
      <br />
      <b>
        筹款总目标：
        1000
      </b>
      <p
        className="friend"
      >
        你的好友:
        Samuel,Tony,Sherry参与了捐赠
        参与了捐赠
      </p>
      <a
        style={
          Object {
            "color": "#00a1ffd9",
            "fontSize": ".9rem",
          }
        }
      >
        查看更多
      </a>
    </div>
  </div>,
  <div
    className="detailBoard"
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": "12px",
        "height": "unset",
      }
    }
  >
    <div
      className="detailTitle"
    >
      项目详情
    </div>
    美国发起了一场针对中国的地缘政治竞争，但没有预先制定协调一致的战略。在中国承受一个世纪的屈辱时，美国则享受了一个世纪的胜利。中国人在思考上是非常谨慎、实用主义而且非常具有战略性的。鉴于美国人130年来赢得了所有竞争，他们本着不可能失败的原则出发而让自己处于危险境地。我这里引用格雷厄姆·艾利森的话，他说美国希望中国跟它一样。一定要注意你所希望的东西啊！当美国在19世纪90年代成为一个强大国家，面对类似中国在今天的局面时，美国人做的第一件事就是向其他国家宣战，比如从西班牙手中掠取了菲律宾。美国还会拿走由越南、菲律宾、马来西亚和文莱控制的所有岛礁。中国拥有可以这么做的军事实力。但中国没有这么做。
  </div>,
  <div
    className="buttonBox"
  >
    <button
      className="button"
      onClick={null}
      style={
        Object {
          "backgroundColor": "#E56862",
        }
      }
    >
      立即捐款
    </button>
    <button
      className="button"
      onClick={null}
      style={
        Object {
          "backgroundColor": "#F59B40",
        }
      }
    >
      进入投票
    </button>
  </div>,
]
`;
